src/main/java/org/candlepin/util/ConsumerFeedFactExtractor.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.util;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ConsumerFeedFactExtractor {
```

Comment:
nit: given that this will be used for exactly one specific operation, it probably shouldn't exist as a general util. I'm not sure there's a ton of value in abstracting it into its own class, but even still, it could exist as a subclass of the consumer feeds query builder.

Reply:
ok moved to the builder